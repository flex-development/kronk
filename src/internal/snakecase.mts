/**
 * @file Internal - snakecase
 * @module kronk/internal/snakecase
 */

/**
 * Convert a kebab-case `string` to snake_case.
 *
 * @internal
 *
 * @this {void}
 *
 * @param {string} string
 *  The string to convert
 * @return {string}
 *  Snake cased `string`
 */
function snakecase(this: void, string: string): string {
  return string.split('-').reduce((acc, part) => {
    return acc + '_' + part.toLowerCase()
  })
}

export default snakecase
