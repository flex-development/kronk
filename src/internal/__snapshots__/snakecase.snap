// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:internal/snakecase > should return snake_case string ("debug") 1`] = `"debug"`;

exports[`unit:internal/snakecase > should return snake_case string ("esbuild.log-level") 1`] = `"esbuild.log_level"`;

exports[`unit:internal/snakecase > should return snake_case string ("logLevel") 1`] = `"logLevel"`;

exports[`unit:internal/snakecase > should return snake_case string ("pizza-type") 1`] = `"pizza_type"`;

exports[`unit:internal/snakecase > should return snake_case string ("resolve.conditions") 1`] = `"resolve.conditions"`;
